### DISCUSSION TITLE
Option Not in Upload

### DISCUSSION THREAD

**Discussion by KDMcMullan** [egzumer/uvk5-chirp-driver#29] [Q&A]
The option to set Quansheng UV-K5 (egzumer) is available in the "Download to" dialog, but is not available in the "Upload to" dialog.

With my apologies, the reason we need a CHIRP driver update is not very clear. If I understood why we need it maybe the question would answer itself.

I'd be happy to update documentation once I'm sufficiently educated! Thanks.

**Comment by joc2** [comment]
The reason you need the egzumer driver is to get all the option is in the firmware egzumer, 
if you take the original driver uv-k5 include in chirp, you will miss the special option not include in the original driver, example, all the extra field in the memori tab, will not show. The programmation of the key M and many more option...

when read the info from the radio (download) , chirp create a tab with all the data from the radio ( number 1 in picture),   
when you want to write the info to the radio (upload).  The information of the radio is all ready in the chirp file, you dont have to select it, it show up filled... ( number 2 in picture)

suggest you to read wiki on egzumer for all option in firmware 

https://github.com/egzumer/uv-k5-firmware-custom/wiki

   
![image](https://github.com/egzumer/uvk5-chirp-driver/assets/56229329/6e99313e-1c9e-4d52-ae3c-4c66ffe4965f)

**Reply by KDMcMullan** [reply]
Yes, thank you. Figured that bit out. Unfortunately, still don't have an answer to the core question as to why the option to set Quansheng UV-K5 (egzumer) is available in the "Download to" dialog, but is not available in the "Upload to" dialog. It doesn't really seem to matter: it seems I can upload anyway! Thanks again.

**Reply by egzumer** [reply]
Configuration is model specific, you cannot change configuration model.
What if it was a configuration downloaded from Yaesu? How do you expect to upload it to Quansheng?
Same thing.

**Reply by KDMcMullan** [reply]
With apologies, the question was in response to the documentation "Choose downloaded uvk5_EGZUMER.py, new radio will appear in Quansheng section in download/upload function." Indeed, it does appear in "download to" but does appear in "upload to".
In fact, on subsequent executions of the program, I have noticed that this only happens once. If I close the dialog and re-open it, the new firmware DOES appear in the upload list.
I'm happy that I have a solution, and no-one else seems to have noticed! Thank you.

**Reply by KDMcMullan** [reply]
For further clarity, if a file was downloaded from a Yashu, I would not expect it to upload to a Quansheng (at least not in its entirety). However, the user may assume (as I did) that this was an opportunity to select their destination radio type.
